iron-location.html

```diff
@@ -181,7 +181,11 @@
             this.path).replace(/\#/g, '%23').replace(/\?/g, '%3F');
         var partiallyEncodedQuery = '';
         if (this.query) {
-          partiallyEncodedQuery = '?' + this.query.replace(/\#/g, '%23');
+          // decoding here needed to prevent encoding the % sign. decodeURIComponent
+          // because it will not decode the hash otherwise. encodeURI and not
+          // component because it will encode ? to %3F otherwise.
```

Comment:
`hostname?foo?bar` is valid when typed into the browser, so I decided not to fight it and leave it that way. Since I'm choosing to keep `?foo?bar` and not `?foo%3Fbar`, if I use `encodeURIComponent` then i'd have to replace `%3F` with `?` instead of using `encodeURI` and replacing `#` wigh `%23`. in other words: because I'd have to replace something anyway